### PR TITLE
Added `InstructionsPanel` component

### PR DESCRIPTION
**Pull request by BrookJeynes**
## Changes
- Added `InstructionsPanel` component
- Added `Tooltip` component
- Added `Level` struct to hold `TerminalContent` and `InstructionsPanel` content
- Added `Levels` struct to hold all level data and provided a function to get a level
- Fixed bug where user could keep entering input in terminal even though the level had ended
- Added mock first level
- Added ability to swap between levels
- Added `class` prop to components to allow style overrides

## Images
image

**Figure: New instructions component on the left**

### CODE REVIEW

File: src/pages/home.rs [hunk 1]
```diff
@@ -21,24 +23,74 @@ pub fn Home(cx: Scope) -> impl IntoView {
                 </div>
             </div>
         }.into_any()),
-        (String::from("help"), view! {cx, <p><span class="text-pastel-blue">"❯ "</span>"help"</p>}.into_any()),
     ]);
 
+    // Callbacks
+    let increment_level = move || {
+        set_level(level() + 1);
```
Comment (miyuushi) [inline src/pages/home.rs]: I would change this to `level.update(|n| *n += 1)` as it's the recommended way to update signals.

File: src/pages/home.rs [hunk 2]
```diff
@@ -21,24 +23,74 @@ pub fn Home(cx: Scope) -> impl IntoView {
...
+    };
+
+    let decrement_level = move || {
+        if level() > 0 {
+            set_level(level() - 1);
```
Comment (miyuushi) [inline src/pages/home.rs]: Same change here, but with a decrement.

File: src/components/terminal.rs [hunk 3]
```diff
@@ -2,13 +2,16 @@ use crate::types::terminal_content::TerminalContent;
...
 use std::time::Duration;
 
+// TODO - Tech Debt: Find a way to not allow the whole
+// component but rather the single closure
+#[allow(clippy::redundant_closure)]
 #[component]
 pub fn Terminal<F>(
     cx: Scope,
-    // TODO - Tech Debt: Find a way to pass terminal in as mut
     terminal: TerminalContent,
```
Comment (miyuushi) [inline src/components/terminal.rs]: Is this supposed to be reactive/change later on? It seems like `TerminalContent` is only for one level, so I would assume this is updated? I would consider passing a `Signal<TerminalContent>` if this is the case.

**Review by miyuushi — COMMENTED** [review]
So, a bit of a suggestion on how you might want to approach the levels: since you already have a `level` signal, what you could do is make a derived signal of type `TerminalContent` that depends on `level`, so for instance you could index from a `Vec<TerminalContent>` with the current `level` value to get the `TerminalContent` of the current level.

File: src/components/terminal.rs [hunk 4]
```diff
@@ -1,21 +1,24 @@
-use crate::types::terminal_content::TerminalContent;
+use crate::types::level::Level;
 use leptos::{ev::SubmitEvent, html::Input, *};
 use std::time::Duration;
 
+// TODO - Tech Debt: Find a way to not allow the whole
+// component but rather the single closure
+#[allow(clippy::redundant_closure)]
```
Comment (BrookJeynes) [inline src/components/terminal.rs]: So many items in Leptos rely on closures to update values, this one might just need to go on all components.

**Review by BenSimmers — APPROVED** [review]
lgtm!

**Comment by miyuushi** [comment]
This seems ok to merge for now

**Pull request merged**